python/openmldb_tool/diagnostic_tool/server_checker.py

```diff
@@ -33,6 +36,58 @@ def check_components(self):
         components_map = self._get_components(show=True)
         return self._check_status(components_map)
 
+    def check_connection(self):
+        """check all compontents connections"""
+        component_map = self._get_components(show=False)
+        t = PrettyTable()
+        t.title = "Connections"
+        t.field_names = ["Endpoint", "version", "cost_time", "extra"]
+        err = ""
+        taskmanager = component_map.pop("taskmanager")  # extract taskmanager
+        other_components = [component for role in component_map.values() for component in role]  # extract other components
+        for (endpoint, _) in other_components:
+            version, response_time, ex, e = self._get_information(endpoint)
+            t.add_row([endpoint, version, response_time, ex])
+            err += e
+        for (endpoint, _) in taskmanager:
+            version, response_time, ex, e = self._get_information_taskmanager(endpoint)
+            t.add_row([endpoint, version, response_time, ex])
+            err += e
+        print(t)
+        if err:
+            print(err)
+
+    def _get_information(self, endpoint):
+        """get informations from components except taskmanager"""
+        try:
+            response = requests.get(f"http://{endpoint}/status", timeout=1)  # the connection timeout is 1 second
+            response.raise_for_status()
+        except Exception as e:
+            err = endpoint + "\n" + str(e) + "\n"
+            return "-", "timeout", "Error: See below", err
+        text = response.text
+        regex = re.compile("version: (.*?)\\n")
+        match = re.search(regex, text)
+        version = match.group(1)
+        ex = ""
+        response_time = str(response.elapsed.microseconds / 1000) + "ms"
+        return version, response_time, ex, ""
+
+    def _get_information_taskmanager(self, endpoint):
+        """get informations from taskmanager"""
+        try:
+            response = requests.post(f"http://{endpoint}/openmldb.taskmanager.TaskManagerServer/GetVersion", json={}, timeout=1)  # the connection timeout is 1 second
```

Comment:
GetVersion rpc may cost ~500ms in local env, it's better no timeout setting here, to avoid timeout in production env or worse network env

Reply:
cancel timeout setting here